docs/reference/images/advanced_topics.rst

```diff
@@ -2,6 +2,82 @@
 Advanced topics
 ===============
 
+.. _custom_image_model:
+
+Custom image model
+==================
+
+The ``Image`` model can be customised, allowing additional fields to be added
+to images.
+
+To do this, you need to add two models to your project:
+
+ - The image model itself that inherits from
+   ``wagtail.wagtailimages.models.AbstractImage``. This is where you would add
+   your additional fields
+ - The renditions model that inherits from
+   ``wagtail.wagtailimages.models.AbstractRendition``. This is used to store
+   renditions for the new model.
+
+Here's an example:
+
+.. code-block:: python
+
+    # models.py
+
+    from django.db import models
+    from wagtail.wagtailimages.models import AbstractImage, AbstractRendition
+
+
+    class CustomImage(AbstractImage):
+        # Add any extra fields to image here
+
+        # eg. To add a caption field:
+        # caption = models.CharField(max_length=255)
+
+
+    class CustomRendition(AbstractRendition):
+        image = models.ForeignKey(CustomImage, related_name='renditions')
+
+        class Meta:
+            unique_together = (
+                ('image', 'filter', 'focal_point_key'),
+            )
+
+
+    # Delete the source image file when an image is deleted
+    @receiver(pre_delete, sender=CustomImage)
+    def image_delete(sender, instance, **kwargs):
+        instance.file.delete(False)
+
+
+    # Delete the rendition image file when a rendition is deleted
+    @receiver(pre_delete, sender=CustomRendition)
+    def rendition_delete(sender, instance, **kwargs):
+        instance.file.delete(False)
+
+.. note::
+
+    If you are using image feature detection, follow these instructions to
+    enable it on your custom image model: :ref:`feature_detection_custom_image_model`
+
+Then set the ``WAGTAILIMAGES_IMAGE_MODEL`` setting to point to it:
+
+.. code-block:: python
+
+    WAGTAILIMAGES_IMAGE_MODEL = 'images.CustomImage'
```

Comment:
This is also ref'd in `howto/settings`. Perhaps the reference in settings should have a link pointing to this page